cli/scorecard/score.go

```diff
@@ -148,19 +150,67 @@ func (inventory *InventoryConfig) Score(config *ScoringConfig) error {
 	err = config.attachViolations(auditResult)
 
 	if len(auditResult.Violations) > 0 {
-		fmt.Printf("\n\n%v total issues found\n", len(auditResult.Violations))
-		for _, category := range config.categories {
-			fmt.Printf("\n\n%v: %v issues found\n", category.Name, category.Count())
-			fmt.Printf("----------\n")
-			for _, cv := range category.constraints {
-				fmt.Printf("%v: %v issues\n", cv.GetName(), cv.Count())
-				for _, v := range cv.Violations {
-					fmt.Printf("- %v\n\n",
-						v.Message,
-					)
-					Log.Debug("Violation metadata", "metadata", v.GetMetadata())
+		content := ""
```

Comment:
Instead of building a string in-memory + then writing it to the chosen destination (which is potentially inefficient), can we *first* build an io.Writer for the destination then have the various outputters write to it directly?

Reply:
addressed in 6c16093